src/chart/XYAxis.js

```diff
@@ -147,6 +170,12 @@
             .target(this.svg.node())
             .guideTarget(this.svgValueGuide.node())
         ;
+        if(this.y2AxisCanShow()){
+           this.valueAxis2
+               .target(this.svg.node())
+               .guideTarget(this.svgValueGuide.node())
+            ;    
+        }
```

Comment:
I think you should always initialize the valueAxis2 - but default it to "hidden" (which is a display option on the Axis Widget)